### PR TITLE
Checks for bad requests and prevents notice

### PR DESCRIPTION
**Pull request by nicholasruunu**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -104,7 +110,7 @@ private function parseHeaders($raw)
      */
```
Comment (Nyholm) [inline src/Client.php:106]: Update return value docs

File: src/Client.php [hunk 2]
```diff
@@ -31,16 +32,21 @@ class Client implements HttpClient
 
     public function __construct()
     {
-        curl_setopt_array($this->curl = curl_init(), self::CURL_DEFAULT_OPTIONS);
+        $this->curl = curl_init();
+
+        if (false === $this->curl) {
+            throw new Exception('Unable to create a new cURL handle');
+        }
```
Comment (Nyholm) [inline src/Client.php:36]: You should use inline variable

File: src/Client.php [hunk 3]
```diff
@@ -31,16 +32,21 @@ class Client implements HttpClient
 
     public function __construct()
     {
-        curl_setopt_array($this->curl = curl_init(), self::CURL_DEFAULT_OPTIONS);
+        $this->curl = curl_init();
+
+        if (false === $this->curl) {
+            throw new Exception('Unable to create a new cURL handle');
```
Comment (Nyholm) [inline src/Client.php:35]: Lets use \Exception

Comment (nicholasruunu) [inline src/Client.php:35]: It's imported in the top already, or do you want it down there?

**Comment by Nyholm** [comment]
Thank you!

**Pull request merged**